src/shmem_internal.h

```diff
@@ -486,39 +486,33 @@ extern uint64_t (*shmem_internal_gettid_fn)(void);
 extern void shmem_internal_register_gettid(uint64_t (*gettid_fn)(void));
 
 static inline
-void shmem_internal_bit_set(void * ptr, size_t size, size_t index)
+void shmem_internal_bit_set(unsigned char *ptr, size_t size, size_t index)
```

Comment:
Would be nice to have a DEBUG function to print out the full bit array at every PE.  This might come in handy when debugging down the line.  (Could handle as a separate issue/PR)

Reply:
I had something like this during development - it's still in the git history - something like:

```
static inline
void shmem_internal_bit_print(const unsigned char *ptr, const size_t size)
{
    /* The following ignores endianess: */
    for (int i = size-1; i >= 0; i--) {
        for (int j = CHAR_BIT-1; j >= 0; j--) {
            unsigned char bit_val = (ptr[i] >> j) & 1;
            printf("%u", bit_val);
        }
    }
    printf("\n");
}
```
I modified it a bit (:wink:) to take an `unsigned char *` instead of `void *`.